streams/service/src/main/java/com/hortonworks/streamline/streams/service/SchemaResource.java

```diff
@@ -132,4 +133,61 @@ public Response getKafkaSourceSchema(@PathParam("topicName") String topicName,
         }
     }
 
+    @GET
+    @Path("/{schemaName}/versions")
+    @Timed
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getAllSchemaVersions(@PathParam("schemaName") String schemaName,
+                                         @Context SecurityContext securityContext) {
+        try {
+            LOG.info("Get all versions for schema : {}", schemaName);
+            Collection<SchemaVersionInfo> schemaVersionInfos = schemaRegistryClient.getAllVersions(schemaName);
+            Collection<String> schemaVersions = new ArrayList<>();
+            if (schemaVersionInfos != null && !schemaVersionInfos.isEmpty()) {
+                LOG.debug("######### Received schema versions from schema registry: {}", schemaVersionInfos);
+                for (SchemaVersionInfo schemaVersionInfo: schemaVersionInfos) {
+                    schemaVersions.add(schemaVersionInfo.getVersion().toString());
+                }
+                return WSUtils.respondEntities(schemaVersions, OK);
+            } else {
+                LOG.debug("######### Received null response for schema versions from schema registry for schema: {}", schemaName);
+                return WSUtils.respondEntity(schemaVersions, NOT_FOUND);
+            }
+        } catch (SchemaNotFoundException e) {
+            LOG.error("Schema not found: [{}]", schemaName, e);
+            throw EntityNotFoundException.byId(schemaName);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @GET
+    @Path("/{schemaName}/versions/{version}")
+    @Timed
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getSchemaForVersion(@PathParam("schemaName") String schemaName, @PathParam("version") String version,
+                                        @Context SecurityContext securityContext) {
+        try {
+            LOG.info("Get schema:version {}", schemaName + ":" + version);
+            SchemaVersionInfo schemaVersionInfo = schemaRegistryClient.getSchemaVersionInfo(new SchemaVersionKey(schemaName, Integer.parseInt
+                    (version)));
+            if (schemaVersionInfo != null) {
+                LOG.debug("######### Received schema version from schema registry: {}", schemaVersionInfo);
+                String schema = schemaVersionInfo != null ? schemaVersionInfo.getSchemaText() : null;
+                if (schema != null && !schema.isEmpty()) {
+                    schema = AvroStreamlineSchemaConverter.convertAvroSchemaToStreamlineSchema(schema);
+                }
+                LOG.debug("######### Converted schema: {}", schema);
+                return WSUtils.respondEntity(schema, OK);
+            } else {
+                return WSUtils.respondEntity("", NOT_FOUND);
```

Comment:
may want to throw `SchemaNotFoundException ` here so that it throws right error to the client.